### PR TITLE
Allow runtimes to provide additional context on CRI pull errors

### PR DESCRIPTION
**Pull request by saschagrunert**
#### What type of PR is this?

/kind cleanup

#### What this PR does / why we need it:
Right now container runtimes have no way to provide additional context to the pull errors. We now loosen the constraints and check for additional messages after the actual CRI errors, which allows to enrich the verbosity of the warning events, for example:

```
Warning  Failed     2s (x3 over 43s)   kubelet            Failed to pull image "localhost:5000/foo": RegistryUnavailable: pinging container registry localhost:5000: Get "http://localhost:5000/v2/": dial tcp [::1]:5000: connect: connection refused
```

#### Which issue(s) this PR fixes:

None

#### Special notes for your reviewer:
Follow-up on: https://github.com/kubernetes/kubernetes/pull/117717
#### Does this PR introduce a user-facing change?

```release-note
None
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

**Comment by saschagrunert** [comment]
PTAL @kubernetes/sig-node-pr-reviews @haircommander @mrunalp

File: pkg/kubelet/images/image_manager.go [hunk 1]
```diff
@@ -172,13 +173,21 @@ func (m *imageManager) EnsureImageExists(ctx context.Context, pod *v1.Pod, conta
...
 		return errMsg, crierrors.ErrRegistryUnavailable
 	}
 
-	if err.Error() == crierrors.ErrSignatureValidationFailed.Error() {
-		errMsg = fmt.Sprintf("image pull failed for %s because the signature validation failed.", container.Image)
+	if strings.HasPrefix(err.Error(), crierrors.ErrSignatureValidationFailed.Error()) {
+		errMsg = fmt.Sprintf(
+			"image pull failed for %s because the signature validation failed%s",
+			container.Image,
+			strings.TrimPrefix(err.Error(), crierrors.ErrSignatureValidationFailed.Error()),
```
[the remark this answers is not in the document]
Comment (saschagrunert) [reply pkg/kubelet/images/image_manager.go:189]: Yes, done :+1:

**Comment by haircommander** [comment]
/lgtm

thanks!

**Review by mrunalp — APPROVED** [review]

**Pull request merged**